auto-start-reverse WPT is failing

karlcow opened the pull request:
#### ced6588f9f23122a7de18471a3e2ec2067bb560c
<pre>
auto-start-reverse WPT is failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=269477">https://bugs.webkit.org/show_bug.cgi?id=269477</a>
<a href="https://rdar.apple.com/123453058">rdar://123453058</a>

Reviewed by Said Abou-Hallawa.

This doesn&apos;t add any new features.
It progresses some WPT tests and aligns the WebKit implementation with
what Firefox is already doing. It also matches the specification IDL by
adding SVG_MARKER_ORIENT_AUTO_START_REVERSE.
The SVG 2 specIDL for SVG Marker Element and its orient type was
recently fixed.
See <a href="https://github.com/w3c/svgwg/pull/994">https://github.com/w3c/svgwg/pull/994</a>
and discussions on <a href="https://github.com/w3c/svgwg/issues/993">https://github.com/w3c/svgwg/issues/993</a>
and <a href="https://github.com/w3c/svgwg/issues/424">https://github.com/w3c/svgwg/issues/424</a>

* LayoutTests/imported/w3c/web-platform-tests/svg/animations/animate-marker-orient-from-auto-to-auto-start-reverse-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGAnimatedEnumeration-SVGMarkerElement-expected.txt:
* LayoutTests/svg/animations/animate-marker-orient-from-angle-to-autostartreverse-expected.txt:
* LayoutTests/svg/animations/animate-marker-orient-from-angle-to-autostartreverse.html:
* LayoutTests/svg/dynamic-updates/SVGMarkerElement-dom-orient-attr-expected.txt:
* LayoutTests/svg/dynamic-updates/SVGMarkerElement-dom-orient-attr.html:
* Source/WebCore/svg/SVGMarkerElement.cpp:
(WebCore::SVGMarkerElement::setOrientToAutoStartReverse):
* Source/WebCore/svg/SVGMarkerElement.h:
* Source/WebCore/svg/SVGMarkerElement.idl:
* Source/WebCore/svg/SVGMarkerTypes.h:
(WebCore::SVGPropertyTraits&lt;SVGMarkerOrientType&gt;::highestEnumValue):

Canonical link: <a href="https://commits.webkit.org/299042@main">https://commits.webkit.org/299042@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c658dd48eebdf95849bea2c9580a59873ff0d88d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117602 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37279 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27904 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/123714 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/69610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/23a5c573-319e-4a98-9cc1-85afaa3799ad) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119480 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37972 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45862 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/123714 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/69610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e70257b3-341e-4faa-9433-deeed5c6b8b7) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120554 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/30222 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/105442 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/123714 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/14f9a7c2-f907-44d3-8039-6d450b9714cc) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/29283 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/23559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/67383 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/99638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/23740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126827 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/44505 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/33481 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/97910 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/44863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/101671 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/97698 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/21027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/40863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18767 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/44376 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/50051 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/43834 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/47182 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45526 "Failed to checkout and rebase branch from PR 49526") | | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/svg/SVGMarkerElement.cpp:
```diff
@@ -166,6 +166,12 @@ void SVGMarkerElement::setOrientToAngle(const SVGAngle& angle)
     invalidateMarkerResource();
 }
 
+void SVGMarkerElement::setOrientToAutoStartReverse()
+{
+    Ref { m_orientType }->setBaseVal(SVGMarkerOrientAutoStartReverse);
+    invalidateMarkerResource();
+}
```
This function is not called. I think `SVGMarkerElement.idl` should have a method to call this function similar to:
```
    undefined setOrientToAuto();
    undefined setOrientToAngle(SVGAngle angle);
```

karlcow replied:
fixed.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1b3caea3e3819ed54d35ded5baafdd4e4c00117c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49526

shallawa commented on Source/WebCore/svg/SVGMarkerElement.idl:
```diff
@@ -48,6 +49,7 @@
 
     undefined setOrientToAuto();
     undefined setOrientToAngle(SVGAngle angle);
+    undefined setOrientToAutoStartReverse();
```
I do not see this method in the spec https://svgwg.org/svg2-draft/painting.html#InterfaceSVGMarkerElement. But it looks like adding the new enum value and adding this method are the only way to get this interface clean and complete. Please create a GitHub issue if there no one exists.

nikolaszimmermann commented:
I was about to comment that I never heard about that API. I did not check in depth, but does the WPT test in-question call that specific function?Am 19.08.2025 um 19:35 schrieb Said Abou-Hallawa ***@***.***>:﻿
@shallawa approved this pull request.



In Source/WebCore/svg/SVGMarkerElement.idl:
> @@ -48,6 +49,7 @@
 
     undefined setOrientToAuto();
     undefined setOrientToAngle(SVGAngle angle);
+    undefined setOrientToAutoStartReverse();

I do not see this method in the spec https://svgwg.org/svg2-draft/painting.html#InterfaceSVGMarkerElement. But it looks like adding the new enum value and adding this method are the only way to get this interface clean and complete. Please create a GitHub issue if there no one exists.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because your review was requested.Message ID: ***@***.***>

karlcow commented:
> I was about to comment that I never heard about that API. I did not check in depth, but does the WPT test in-question call that specific function?

It doesn't.

karlcow committed:
auto-start-reverse WPT is failing
https://bugs.webkit.org/show_bug.cgi?id=269477
rdar://123453058

Reviewed by Said Abou-Hallawa.

This doesn't add any new features.
It progresses some WPT tests and aligns the WebKit implementation with
what Firefox is already doing. It also matches the specification IDL by
adding SVG_MARKER_ORIENT_AUTO_START_REVERSE.
The SVG 2 specIDL for SVG Marker Element and its orient type was
recently fixed.
See https://github.com/w3c/svgwg/pull/994
and discussions on https://github.com/w3c/svgwg/issues/993
and https://github.com/w3c/svgwg/issues/424

* LayoutTests/imported/w3c/web-platform-tests/svg/animations/animate-marker-orient-from-auto-to-auto-start-reverse-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGAnimatedEnumeration-SVGMarkerElement-expected.txt:
* LayoutTests/svg/animations/animate-marker-orient-from-angle-to-autostartreverse-expected.txt:
* LayoutTests/svg/animations/animate-marker-orient-from-angle-to-autostartreverse.html:
* LayoutTests/svg/dynamic-updates/SVGMarkerElement-dom-orient-attr-expected.txt:
* LayoutTests/svg/dynamic-updates/SVGMarkerElement-dom-orient-attr.html:
* Source/WebCore/svg/SVGMarkerElement.cpp:
(WebCore::SVGMarkerElement::setOrientToAutoStartReverse):
* Source/WebCore/svg/SVGMarkerElement.h:
* Source/WebCore/svg/SVGMarkerElement.idl:
* Source/WebCore/svg/SVGMarkerTypes.h:
(WebCore::SVGPropertyTraits<SVGMarkerOrientType>::highestEnumValue):

Canonical link: https://commits.webkit.org/299042@main

webkit-commit-queue commented:
Committed 299042@main (ced6588f9f23): <https://commits.webkit.org/299042@main>

Reviewed commits have been landed. Closing PR #49526 and removing active labels.
